crates/turborepo-lib/src/gitignore.rs

```diff
@@ -0,0 +1,141 @@
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+use std::{
+    fs::{File, OpenOptions},
+    io,
+    io::{BufRead, Write},
+};
+
+use turbopath::AbsoluteSystemPathBuf;
+
+const TURBO_GITIGNORE_ENTRY: &str = ".turbo";
+const GITIGNORE_FILE: &str = ".gitignore";
+
+pub fn ensure_turbo_is_gitignored(repo_root: &AbsoluteSystemPathBuf) -> Result<(), io::Error> {
+    let gitignore_path = repo_root.join_component(GITIGNORE_FILE);
+
+    if !gitignore_path.exists() {
```

Comment:
We should generally prefer `try_exists` over `exists` as the latter will report `false` if we can't verify the path exists due to permission issues. [docs](https://doc.rust-lang.org/std/fs/fn.try_exists.html)
```suggestion
    if !gitignore_path.try_exists()? {
```